src/main/java/io/axoniq/axonserver/connector/event/EventChannel.java

```diff
@@ -341,16 +341,46 @@ AggregateEventStream loadSnapshots(String aggregateIdentifier,
      * @param liveStream      whether to continue processing live events
      * @return a ResultStream containing the query results
      */
-    ResultStream<EventQueryResultEntry> queryEvents(String queryExpression, boolean liveStream);
+    default ResultStream<EventQueryResultEntry> queryEvents(String queryExpression, boolean liveStream) {
+        return queryEvents(queryExpression, liveStream, null);
+    }
 
+    /**
+     * Queries the Event Store for events using given {@code queryExpression}. The given {@code liveStream} indicates
+     * whether the query should complete when the end of the Event Stream is reached, or if the query should continue
+     * processing events as they are stored.
+     * When a {@code contextName} is provided the query is executed for that context, instead of the main context for
+     * the channel.
+     *
+     * @param queryExpression a valid Event Stream Query Language expression
+     * @param liveStream      whether to continue processing live events
+     * @param contextName     the name of the context of the event store to query
+     * @return a ResultStream containing the query results
+     */
+    ResultStream<EventQueryResultEntry> queryEvents(String queryExpression, boolean liveStream, String contextName);
```

Comment:
Simply adding this operation is a breaking change for users. It's better if we make this new method `default` and leave the old one as is.